dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/PushAnalysisController.java

```diff
@@ -103,10 +103,20 @@ public void renderPushAnalytics(
 
         logger.info(
             "User '" + currentUserService.getCurrentUser().getUsername() + "' started PushAnalysis for 'rendering'." );
-        String result = pushAnalysisService.generateHtmlReport( pushAnalysis, currentUserService.getCurrentUser(), null );
 
-        response.getWriter().write( result );
-        response.getWriter().close();
+        try
+        {
+            String result = pushAnalysisService.generateHtmlReport( pushAnalysis, currentUserService.getCurrentUser(), null );
+            response.getWriter().write( result );
+            response.getWriter().close();
+
+        }
+        catch ( Exception e )
+        {
+            logger.error( "Failed to render push analysis" );
+            e.printStackTrace();
```

Comment:
Please don't print stack traces. This can create noise in logs. Log the exception `logger.error( "Failed to render push analysis", e );` https://www.slf4j.org/faq.html#exception_message

This way implementers, devs can configure log4j2 how to log this event or to not log it at all 😄